pkg/kotsadm/minio.go

```diff
@@ -97,12 +121,20 @@ func ensureMinioStatefulset(deployOptions types.DeployOptions, clientset *kubern
 	existingMinio.Spec.Template.Spec.Volumes = desiredMinio.Spec.Template.Spec.DeepCopy().Volumes
 	existingMinio.Spec.Template.Spec.Containers[0].Image = desiredMinio.Spec.Template.Spec.Containers[0].Image
 	existingMinio.Spec.Template.Spec.Containers[0].VolumeMounts = desiredMinio.Spec.Template.Spec.Containers[0].DeepCopy().VolumeMounts
+	existingMinio.Spec.Template.Spec.InitContainers = desiredMinio.Spec.Template.Spec.DeepCopy().InitContainers
```

Comment:
what if the kots upgrade command happens to fails after updating the minio statefulset and the migration failed or didn't finish yet, and the user reruns the upgrade command? the init container would be removed because the current minio image in the cluster is the new one, no?

Reply:
Hmm, could maybe leverage the task logic to keep track of the status? I'll think about this more

Reply:
couldn't utilize existing store logic since it depends on kotsadm variables like `util.PodNamespace`, so creating/updating tasks from cli-only would require changes.  instead opted for a simpler implementation with a "status" configmap that could also aid in debugging.